contracts/stabilizer/PegStabilityModule.sol

```diff
@@ -0,0 +1,276 @@
+        reservesThreshold = newReservesThreshold;
+
+        emit ReservesThresholdChanged(oldReservesThreshold, newReservesThreshold);
+    }
+
+    /// @notice helper function to set the target
+    function _setTarget(IPCVDeposit newTarget) internal {
+        require(address(newTarget) != address(0), "PegStabilityModule: Invalid new target");
+        IPCVDeposit oldTarget = target;
+
+        emit TargetChanged(oldTarget, newTarget);
+    }
+
+    /// @notice function to redeem FEI for an underlying asset
+    function redeem(address to, uint256 amountFeiIn) external virtual override nonReentrant whenNotPaused returns (uint256 amountOut) {
+        updateOracle();
+        Decimal.D256 memory price;
+
+        (amountOut, price) = _getRedeemAmountOutAndPrice(amountFeiIn);
+        require(_validPrice(price), "PegStabilityModule: price out of bounds");
+        FEI.transferFrom(msg.sender, address(this), amountFeiIn);
+        FEI.burn(amountFeiIn);
+
+        token.safeTransfer(to, amountOut);
+
+        emit Redeem(to, amountFeiIn);
+    }
+
+    /// @notice function to buy FEI for an underlying asset
+    function mint(address to, uint256 amountIn) external virtual override payable nonReentrant whenNotPaused returns (uint256 amountFeiOut) {
+        require(msg.value == 0, "PegStabilityModule: cannot send eth to mint");
+
+        updateOracle();
+        Decimal.D256 memory price;
+
+        (amountFeiOut, price) = _getMintAmountOutAndPrice(amountIn);
+        require(_validPrice(price), "PegStabilityModule: price out of bounds");
+
+        token.safeTransferFrom(msg.sender, address(this), amountIn);
+
+        _mintFei(msg.sender, amountFeiOut);
+
+        emit Mint(to, amountIn);
+    }
+
+    /// @notice function to pause the PSM. Only accessible in the ERC20PSM
+    /// should be called whenever the DAI price is above $1.02 or below $0.98 as this will pause the contract
+    /// then a guardian or governor will have to unpause the contract
+    function oracleErrorPause() external virtual whenNotPaused {
+        updateOracle();
+        Decimal.D256 memory price = readOracle();
+
+        require(!_validPrice(price), "PegStabilityModule: price not out of bounds");
+
+        _pause();
+    }
+
+    /// @notice helper function to determine if price is within a valid range
+    /// function is private so that classes that inherit cannot use
+    function _validPrice(Decimal.D256 memory price) private pure returns(bool valid) {
+        Decimal.D256 memory floorPrice = Decimal.D256({ value: 0.98 ether });
+        Decimal.D256 memory ceilingPrice = Decimal.D256({ value: 1.02 ether });
+
+        valid = price.greaterThan(floorPrice) && price.lessThan(ceilingPrice);
+        return valid;
+    }
+
+    function _getMintAmountOutAndPrice(uint256 amountIn) private view returns (uint256 amountFeiOut, Decimal.D256 memory price) {
+        price = readOracle();
+        Decimal.D256 memory feiValueOfAmountIn = price.mul(amountIn);
```

Comment:
technically this is usdValue, what naming conventions should we use to keep consistency between mint/redeem because right now redeem is calling this "AdjustedAmountIn"